harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -229,3 +229,45 @@ def test_download_code_writes_output_to_file(
     with open(output_file, "rb") as f:
         file_content = f.read()
         assert file_content == b"b64TgzResponse"
+
+
+@mock.patch("determined.common.api.bindings.get_GetExperimentCheckpoints")
+@pytest.mark.parametrize(
+    "sort_by", [checkpoint.CheckpointSortBy.SEARCHER_METRIC, "val_metric_name"]
+)
+def test_list_checkpoints_calls_bindings_with_params(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+    sort_by: Union[checkpoint.CheckpointSortBy, str],
+) -> None:
+    expref = make_expref(1)
+    ckpt_resp = api_responses.sample_get_experiment_checkpoints()
+    mock_bindings.side_effect = api_responses.iter_pages(
+        pageable_resp=ckpt_resp,
+        pageable_attribute="checkpoints",
+    )
+
+    expref.list_checkpoints(sort_by=sort_by, order_by=checkpoint.CheckpointOrderBy.ASC, limit=5)
+
+    _, call_kwargs = mock_bindings.call_args_list[0]
+
+    assert call_kwargs["orderBy"] == checkpoint.CheckpointOrderBy.ASC._to_bindings()
+    assert call_kwargs["limit"] == 5
+
+    if isinstance(sort_by, checkpoint.CheckpointSortBy):
+        assert call_kwargs["sortByAttr"] == sort_by._to_bindings()
+    else:
+        assert call_kwargs["sortByMetric"] == sort_by if isinstance(sort_by, str) else None
+
+
+def test_list_checkpoints_errors_on_invalid_argument_set(
```

Comment:
I think this is better as two tests. Both because "Invalid argument set" is hard for an oncall to parse, and each test done separately would be easier to immediately understand and because this is testing two different behaviors.